### PR TITLE
Add an interface to set wave size

### PR DESCRIPTION
**Pull request by xuechen417**
Add setShaderWaveSize(ShaderStage, unsigned)

### CODE REVIEW

File: lgc/include/lgc/state/PipelineState.h [hunk 1]
```diff
@@ -269,6 +269,9 @@ class PipelineState final : public Pipeline {
   // Set the default wave size for the specified shader stage
   void setShaderDefaultWaveSize(ShaderStage stage);
 
+  // Set the wave size for the specified shader stage
+  void setShaderWaveSize(ShaderStage stage, unsigned waveSize) { m_waveSize[stage] = waveSize; }
```
Comment (kuhar) [inline lgc/include/lgc/state/PipelineState.h:273]: Do all wave sizes make sense? I thought that only 32 and 64 were currently allowed. If there some limitations, could we assert that nobody passes a bad wave size by accident?

**Comment by amdvlk-admin** [comment]
### Test summary for commit df7d997c9ce607e2d56927924c79a86b5e54b1c5
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev c9d1904b35ac5eb70bd4738a5bf5e537adbf5ab0</li>
<li>SPVGEN: dev 4846370016de9775b7518b51c08260387a7367c4</li>
<li>XGL: dev dd2834ff77dcb40511a18ae0ca678fafe8a339fe</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 570c8bd44f15e3a0ff7547c0804a08549c7fbb3f</li>
</ul></details>

<details><summary>CTS tests (Failed: 3/119542)</summary><ul><li>Built with version 1.2.5.1</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30147/59723 (50.5%)</li>
<li><details><summary>Failed: 1/59723 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.pipeline.creation_feedback.graphics_tests.vertex_stage_fragment_stage
Stack trace: Flake
```
</p></details></li>

<li>Skipped: 29575/59723 (49.5%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Timeout or crash</li>

</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 29248/59819 (48.9%)</li>
<li><details><summary>Failed: 2/59819 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.pipeline.creation_feedback.graphics_tests.vertex_stage_fragment_stage_delayed_destroy
Stack trace: Fail

FAILURE: dEQP-VK.synchronization2.timeline_semaphore.wait.poll_signal_from_device
Stack trace: Crash
```
</p></details></li>

<li>Skipped: 30569/59819 (51.1%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Timeout or crash</li>

</ul></details>


</ul></details>

File: lgc/include/lgc/state/PipelineState.h (hunk 1 above)
Comment (xuechen417) [reply lgc/include/lgc/state/PipelineState.h:273]: Yes, you're right. Fixed.

**Comment by xuechen417** [comment]
V1: Add assertion for the size limits of waves.

File: lgc/include/lgc/state/PipelineState.h (hunk 1 above)
Comment (amdrexu) [reply lgc/include/lgc/state/PipelineState.h:273]: Please add asserts `assert(waveSize == 32 || waveSize == 64)`

Comment (xuechen417) [reply lgc/include/lgc/state/PipelineState.h]: Done.

**Review by kuhar — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

**Pull request merged**